.github/workflows/build-test.yml

```
@@ -26,38 +29,25 @@ jobs:
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
-    
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install setuptools wheel twine
-    
-    - name: Build and install wheel
-      run: |
-        python setup.py sdist bdist_wheel
-        pip install dist/hvac-*.whl
-
-  build-windows:
-    runs-on: windows-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
 
-    steps:
-    - uses: actions/checkout@v3
+    - name: Install Poetry
+      uses: snok/install-poetry@v1.3.1
+      with:
+        version: 1.1.14
+        virtualenvs-create: true
+        virtualenvs-in-project: true
 
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v4
+    - name: Load cached venv
+      id: cached-pip-wheels
+      uses: actions/cache@v2
       with:
-        python-version: ${{ matrix.python-version }}
-    
+        path: ~/.cache
+        key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+
     - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install setuptools wheel twine
+      run: poetry install --no-interaction --no-root
     
-    - name: Build and install wheel
+    - name: Build and install wheel for *nix
```

nit: this is building cross-platform isn't it? I'd also be fine with removing the for/on portion from the name completely
```suggestion
    - name: Build and install wheel on ${{ matrix.os }}
```

---

Whoops, I left this in when I was still using different workflows for *nix and Windows. To use the poetry installer I have to use Bash for both, which allowed me to merge the workflows. I just removed "for *nix".